app/models/katello/concerns/pulp_database_unit.rb

```diff
@@ -51,7 +51,10 @@ def sync_repository_associations(repository, unit_uuids, additive = false)
         table_name = self.repository_association_class.table_name
         attribute_name = "#{self.name.demodulize.underscore}_id"
 
-        existing_ids = self.repository_association_class.where(:repository_id => repository).pluck(attribute_name)
+        existing_associations = self.repository_association_class.where(:repository_id => repository)
+        existing_associations.reload
```

Comment:
I was actually meaning something more like:

```
repository.rpms.reload
```

I'm not sure that we know that association name generically, but maybe this would work:
```
  repository.send(self.class.name.pluralize.to_sym).reload
```